### PR TITLE
[#430] #IMPLEMENT 'assemblyName: DotNet.Testcontainers; f…

### PR DESCRIPTION
**Pull request by s-rech**
…unction: CopyFileFromContainerAsync'

{Implemented CopyFileFromContainerAsync to get files from inside a running container.}

### CODE REVIEW

**Review by HofmeisterAn — COMMENTED** [review]
Thanks. I'll do the review today or tomorrow. I have a few things to catch up on.

File: src/DotNet.Testcontainers/Clients/TestcontainersClient.cs [hunk 1]
```diff
@@ -192,6 +192,44 @@ await this.containers.ExtractArchiveToContainerAsync(id, "/", memStream, ct)
...
+        throw new InvalidOperationException(
+          $"Not possible to get archive with file {filePath} from container: {ex.ResponseBody}",
+          ex);
+      }
+
+      using (var tarInputStream = new TarInputStream(archive, Encoding.Default))
+      {
+        tarInputStream.IsStreamOwner = true;
+        var entry = tarInputStream.GetNextEntry();
+        if (entry == null || entry.IsDirectory || entry.Name != Path.GetFileName(filePath))
```
Comment (HofmeisterAn) [inline src/DotNet.Testcontainers/Clients/TestcontainersClient.cs:215]: I've added some minor improvements. Before I'll merge the pull request, I'd like to ask if this couldn't be simplified to:

```csharp
if (entry.IsDirectory)
{
  throw new InvalidOperationException("Can not read from a directory. Use a file instead.");
}
```

I think we don't need the other conditions. If the file (path) does not exist, `GetArchiveFromContainerAsync` will already fail.

**Review by HofmeisterAn — APPROVED** [review]
I've renamed `CopyFileFromContainerAsync` to `ReadFileAsync` (maybe down- and upload is better in the future) and moved the tests into their own class. Other than that, your pull request looks good! Thanks again.

File: src/DotNet.Testcontainers/Clients/TestcontainersClient.cs (hunk 1 above)
Comment (s-rech) [reply src/DotNet.Testcontainers/Clients/TestcontainersClient.cs:215]: Yes, that is correct.

**Comment by HofmeisterAn** [comment]
Published in `1.6.0-beta.2232`, closes #430.

**Pull request closed without merging**